Kernel/System/Console/Command/Maint/Elasticsearch/Migration.pm

```diff
@@ -690,6 +696,9 @@ sub _IndexSettingsGet {
     }
 
     my $IndexSettingsTemplate = $Kernel::OM->Get('Kernel::Config')->Get('Elasticsearch::IndexSettings');
```

Comment:
This is wrong as it is. Generally there is a problem with the naming I was not aware of until now. You are using Elasticsearch::IndexSettings as the new ElasticsearchArticleIndexCreationSettings, which I agree is the better name, as we use it for various indices. In the initial pull request of Tomohisa though, the same name was used for a different setting which is just present in Kernel/Config/Default.pm and can just be changed in Kernel/Config.pm. This is the template for the actual index:

`    $Self->{'Elasticsearch::IndexSettings'} = {
        index => {
            number_of_shards   => '[% Data.NS | uri %]',
            number_of_replicas => '[% Data.NR | uri %]',
        },
        'index.mapping.total_fields.limit' => '[% Data.FieldsLimit | uri %]',
    };`

I would suggest to rename the template to Elasticsearch::IndexTemplate###Default. People should then be able to also put specific templates as Elasticsearch::IndexTemplate###Ticket, or Customer,... to Kernel/Config.pm. (This should be noted in a comment in Default.pm.)

Reply:
Ah, sorry, the formatting here is bad. Please have a look for Elasticsearch::IndexSettings in Kernel/Config/Default.pm.